ui/src/search/helper.js

```diff
@@ -32,6 +42,35 @@ export default class SearchHelper {
     );
   }
 
+  updateLocation(urlSuffix = '') {
+    const searchUrl = `${this.getPathname()}?${this.getQueryString()}${urlSuffix}`;
+    if (this.isInitialQueryUpdate() || this.dueToNavigationToSearchPage) {
```

Comment:
So in the end we check for ```initialQueryUpdate``` even if it is embedded search? Like I said, it is worth it to be able to get rid of the code duplication. It looks better :+1:

Reply:
no we don't.. embedded ones don't update location (yet)